### PR TITLE
Fix umask option bug(issues#1622)

### PR DESCRIPTION
**Pull request by hramezani**
Fix umask option bug. [issue_1622](https://github.com/benoitc/gunicorn/issues/1622l)

### CODE REVIEW

File: gunicorn/config.py [hunk 1]
```diff
@@ -43,8 +43,8 @@ def make_settings(ignore=None):
 
 
 def auto_int(_, x):
-    if x.startswith('0') and not x.lower().startswith('0x'):
-        # for compatible with octal numbers in python3
+    # for compatible with octal numbers in python3
+    if re.match('0(\d)', x, re.IGNORECASE):
```
Comment (berkerpeksag) [inline gunicorn/config.py]: I'd use `r''` to avoid subtle bugs.

**Comment by hramezani** [comment]
@berkerpeksag thanks, I add those examples to `test_umask_config`.

**Comment by berkerpeksag** [comment]
I think the problem here is that our test code doesn't call `auto_int` so even if we already added a test for `--umask 0` we missed this bug.

**Comment by berkerpeksag** [comment]
Nevermind, the existing test was wrong.

File: tests/test_config.py [hunk 2]
```diff
@@ -375,6 +375,8 @@ def test_reload(options, expected):
 
 @pytest.mark.parametrize("options, expected", [
     (["--umask 0", "myapp:app"], 0),
```
Comment (berkerpeksag) [inline tests/test_config.py]: This should read:

```py
(["--umask", "0", "myapp:app"], 0),
```

File: tests/test_config.py [hunk 3]
```diff
@@ -375,6 +375,8 @@ def test_reload(options, expected):
...
+    (["--umask 0o0", "myapp:app"], 0),
```
Comment (berkerpeksag) [inline tests/test_config.py]: Same as above:

```py
(["--umask", "0o0", "myapp:app"], 0),
(["--umask", "0x0", "myapp:app"], 0),
```

**Comment by hramezani** [comment]
@berkerpeksag, thanks, you are right. I fixed the tests

**Comment by berkerpeksag** [comment]
Thanks!

**Pull request merged**